src/components/Form/TimeoutWarning/TimeoutWarning.jsx

```diff
@@ -1,146 +1,84 @@
-}
 
-export const minutes = ms => {
-  return parseInt(roundUp(ms / (60 * 1000), 1), 10)
-}
+import { i18n } from 'config'
 
-export const seconds = ms => {
-  return parseInt(roundUp(ms / 1000, 1), 10)
-}
+import { renewSession } from 'actions/AuthActions'
+
+import { saveSection } from 'components/SavedIndicator/persistence-helpers'
+
+import Modal from 'components/Form/Modal'
+import Svg from 'components/Form/Svg'
 
 export class TimeoutWarning extends React.Component {
   constructor(props) {
     super(props)
+
     this.state = {
-      showWarning: this.props.showWarning,
-      interval: props.interval || 60 * 1000,
-      countdown: 60
+      countdown: 60,
     }
-    this.onDismiss = this.onDismiss.bind(this)
-    this.onClick = this.onClick.bind(this)
-    this.tick = this.tick.bind(this)
+
+    this.interval = 1000 // tick every second
     this.timer = null
   }
 
   componentDidMount() {
     this.resetInterval()
+    this.autosave()
   }
 
   componentWillUnmount() {
     window.clearInterval(this.timer)
   }
 
-  tick() {
-    const now = new Date().getTime()
-    const last = this.props.lastRefresh
-    const timeoutInMilliseconds = this.props.timeout * 60 * 1000
-    const diff = now - last
-    if (seconds(diff) >= seconds(timeoutInMilliseconds)) {
-      // Session has expired
-      this.props.dispatch(tokenError())
-    } else if (this.state.showWarning) {
-      // Ticks are to count down by seconds now
-      this.setState({ countdown: seconds(timeoutInMilliseconds - diff) })
-    } else {
-      // Ticks should be checking if in the last minute of the session
-      if (minutes(diff) >= this.props.timeout - 1) {
-        const application = this.props.app
-        const section = this.props.section.section
-        const subsection = this.props.section.subsection
-        const dispatcher = this.props.dispatch
-        this.setState(
-          {
-            showWarning: true,
-            interval: 1000,
-            countdown: seconds(timeoutInMilliseconds - diff)
-          },
-          () => {
-            // Save the form
-            saveSection(application, section, subsection, dispatcher).catch(
-              error => {
-                alert(error)
-              }
-            )
-            // Change the timer interval
-            this.resetInterval()
-          }
-        )
-      }
-    }
+  onClick = () => {
+    const { dispatch } = this.props
+    dispatch(renewSession())
   }
 
-  onDismiss() {
-    this.refreshToken()
-  }
+  autosave = () => {
+    const { application, section, dispatch } = this.props
 
-  onClick() {
-    this.setState(
-      { showWarning: false, interval: 60 * 1000, countdown: 60 },
-      () => {
-        // Change the timer interval
-        this.resetInterval()
-        // Refresh the token
-        this.refreshToken()
-      }
-    )
+    // Save the form
+    saveSection(application, section.section, section.subsection, dispatch)
+    // TODO error handling
```

Comment:
Does this need to happen for this PR, @suzubara ?

Reply:
No, there wasn't any error handling in place to begin with here. I am going to make a followup issue for converting the saveSection flow into a saga to handle this.